x-pack/metricbeat/module/meraki/device_health/devices.go

```diff
@@ -0,0 +1,231 @@
+	// There are two ways to get this information from the API.
+	// An alternative to this would be to use `/organizations/{organizationId}/wireless/devices/channelUtilization/byDevice`,
+	// avoids the need to extract the filtered network IDs below.
+	// However, the SDK's implementation of that operation doesn't have proper type handling, so we perfer this one.
+	// (The naming is also a bit different in the returned data, e.g. wifi0/wifi1 vs band 2.4/5; 80211/non80211 vs wifi/nonwifi)
+
+	networkIDs := make(map[string]bool)
+	for _, device := range devices {
+		if device.details.ProductType != "wireless" {
+			continue
+		}
+
+		if _, ok := networkIDs[device.details.NetworkID]; !ok {
+			networkIDs[device.details.NetworkID] = true
+		}
+	}
+
+	for networkID := range networkIDs {
+		val, res, err := client.Networks.GetNetworkNetworkHealthChannelUtilization(
+			networkID,
+			&meraki.GetNetworkNetworkHealthChannelUtilizationQueryParams{
+				Timespan: period.Seconds(),
+			},
+		)
+
+		if err != nil {
+			if strings.Contains(string(res.Body()), "MR 27.0") {
+				// "This endpoint is only available for networks on MR 27.0 or above."
+				continue
+			}
+
+			return fmt.Errorf("GetNetworkNetworkHealthChannelUtilization failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+		}
+
+		for _, utilization := range *val {
+			if device, ok := devices[Serial(utilization.Serial)]; ok {
+				if utilization.Wifi0 != nil && len(*utilization.Wifi0) != 0 {
+					// only take the first bucket - collection intervals which result in multiple buckets are not supported
+					device.wifi0.Utilization80211 = (*utilization.Wifi0)[0].Utilization80211
+					device.wifi0.UtilizationNon80211 = (*utilization.Wifi0)[0].UtilizationNon80211
+					device.wifi0.UtilizationTotal = (*utilization.Wifi0)[0].UtilizationTotal
+				}
+				if utilization.Wifi1 != nil && len(*utilization.Wifi1) != 0 {
+					device.wifi1.Utilization80211 = (*utilization.Wifi1)[0].Utilization80211
+					device.wifi1.UtilizationNon80211 = (*utilization.Wifi1)[0].UtilizationNon80211
+					device.wifi1.UtilizationTotal = (*utilization.Wifi1)[0].UtilizationTotal
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func getDeviceLicenses(client *meraki.Client, organizationID string, devices map[Serial]*Device) error {
+	val, res, err := client.Organizations.GetOrganizationLicenses(organizationID, &meraki.GetOrganizationLicensesQueryParams{})
+	if err != nil {
+		return fmt.Errorf("GetOrganizationLicenses failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+	}
+
+	for _, license := range *val {
+		if device, ok := devices[Serial(license.DeviceSerial)]; ok {
+			device.license = &license
+		}
+	}
+
+	return nil
+}
+
+func deviceDetailsToMapstr(details *meraki.ResponseItemOrganizationsGetOrganizationDevices) mapstr.M {
+	return mapstr.M{
+		"device.serial":       details.Serial,
+		"device.address":      details.Address,
```

Comment:
Address and firmware are not added in fields.yml

Reply:
good spot thanks